### PR TITLE
Success message on editing profile picture.

### LINKED ISSUE #6074: Reporting success on profile picture upload/change.
Currently when we upload or change our profile picture, there is no message either if the upload was successful or not.
We could add a message to report that to the user.
Screenshot 2019-07-28 at 1 27 02 AM

### PR DESCRIPTION
**Pull request by Gagan6730**
Fixes #6074 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Screenshot 2019-08-08 at 7 22 11 PMScreenshot 2019-08-08 at 7 24 07 PM

success

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Gagan6730** [comment]
@gautamig54 could you review this?

File: app/views/users/_photo.html.erb [hunk 1]
```diff
@@ -74,9 +84,12 @@
       $('.profile-uploading').hide()
       $('#profile-photo')[0].src = data.result.url
       $('#profile-photo').show()
+      $('#success-upload').css("display","block")
```
Comment (newbazz) [inline app/views/users/_photo.html.erb]: please be uniform with the coding style, there is a space after cooma in one of the changes and there is no space here.
Great Work Indeed

File: app/views/users/_photo.html.erb [hunk 2]
```diff
@@ -74,9 +84,12 @@
...
+      
     },
     // see callbacks at https://github.com/blueimp/jQuery-File-Upload/wiki/Options
     fileuploadfail: function(e,data) {
+      $('#fail-upload').css("display", "block")
```
Comment (newbazz) [inline app/views/users/_photo.html.erb]: here there is a space after the comma, add space in the prev file also.

**Comment by Gagan6730** [comment]
@jywarren Can you have a look at it?

File: app/views/users/_photo.html.erb (hunk 1 above)
Comment (Gagan6730) [reply app/views/users/_photo.html.erb]: @newbazz I have removed the space. I will take care of these kind of things from now on.

**Comment by SidharthBansal** [comment]
@Gagan6730 please try some of the other unresolved issues to help us.
Thanks Gagan

**Comment by Gagan6730** [comment]
@SidharthBansal @gauravano I have also resolved the bug where when the profile pic is uploaded, it did not show up in the "profile pic edit area"(shown below) until the page was reloaded. 
Screenshot 2019-08-10 at 7 58 48 PM

So should I commit the resolved code in this pull request or have another on for the same?

Thanks

**Comment by gauravano** [comment]
>  I have also resolved the bug where when the profile pic is uploaded, it does not show up in the "profile pic edit area"(shown below) until the page was reloaded.

That's great! Please commit on this PR.

**Comment by gauravano** [comment]
Also, once the upload is done, the success div looks good but we can remove it after 5/10 sec. Also, you can also show alert using NotyJS(it's already configured), you can use it directly. Example: https://github.com/publiclab/plots2/blob/351ff5231406260a3cea62035453edd5d1b1dc4c/app/views/relationships/create.js.erb#L2

Great work!!

**Comment by Gagan6730** [comment]
> > I have also resolved the bug where when the profile pic is uploaded, it does not show up in the "profile pic edit area"(shown below) until the page was reloaded.
> 
> That's great! Please commit on this PR.

@gauravano right now, what I have done is when we upload, the progress bar shows up and when it is complete, the page reloads and shows the success message along with updated profile picture. Will this work? or do I need to figure another way where the page does not reload.
Thanks

**Comment by gauravano** [comment]
You can show alert without page reload. For image preview, it will appear on reload, but yeah, we should try to show preview without reload.

**Comment by Gagan6730** [comment]
> You can show alert without page reload. For image preview, it will appear on reload, but yeah, we should try to show preview without reload.

I was able to manage the alert without page reload, but the right now I am not able to update the image in the drop area without the reload.

**Comment by gauravano** [comment]
ok, so do you want to give one more try at showing image preview asynchronously?

**Comment by Gagan6730** [comment]
> ok, so do you want to give one more try at showing image preview asynchronously?

Sure, I will give it a try.

**Comment by Gagan6730** [comment]
> You can show alert without page reload. For image preview, it will appear on reload, but yeah, we should try to show preview without reload.

@gauravano I have pushed the changes. Now the profile photo updates without reload and the success message appears as you suggested, using NotyJs and disappears in 3 seconds. 
Please have a look.
Thanks
succes_upload

**Comment by jywarren** [comment]
Awesome work, thank you so much!!!

**Pull request merged**